File: InventoryControlSystem.DAL/Models/Inventory.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace InventoryControlSystem.DAL.Models
{
    public class Inventory
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }
        
        public string Description { get; set; }

        public int UnitCount { get; set; }
        
        public decimal UnitPrice { get; set; }

        public int ReorderCount { get; set; }

        public string AddedBy { get; set; }

        public string AddedByName { get; set; }
        
        [Column(TypeName = "datetime2")]
        public DateTime AddedDate { get; set; }

        public string UpdatedBy { get; set; }

        public string UpdatedName { get; set; }

        [Column(TypeName = "datetime2")]
        public DateTime UpdatedDate { get; set; }
    }
}

File: InventoryControlSystem.UI/src/app/models/inventory.ts
export default class Inventory {
  id?: number;
  name: string;
  description?: string;
  unitCount: number;
  reorderCount: number;
  unitPrice: number;
  addedBy: string;
  addedByName: string;
  addedDate: Date;
  updatedBy?: string;
  updatedByName?: string;
  updatedDate?: Date;
}
